quiz.js

```diff
@@ -22,25 +23,52 @@ var char = document.getElementById("char");
 };
 
 submit.addEventListener("click" , tree)
-
-char.addEventListener("keyup", function(event) {
-	var newRows = enter.value;
+rows.addEventListener("keyup", function(keyPress) {
+	var newRows = rows.value;
 	var newChar = char.value;
 	event.preventDefault();
-    if (newRows.length > 1 || newChar.length > 1) {
+    	    
+  if(keyPress.keyCode < 48 && keyPress.keyCode !== 8 && keyPress.keyCode !== 13 || keyPress.keyCode > 57 && keyPress.keyCode !== 8 && keyPress.keyCode !== 13 && keyPress.keyCode !== 9) {
+        alert("Enter only integers, plz.");
+    }
+
+    else if (newRows.length > 1 || newChar.length > 1) {
     	alert("ONLY ONE CHARACTER");
-    	    }
+
+	}
      else if (event.keyCode == 13) {
         document.getElementById("submit").click();
     }
+   });
 
-    else if(event.keyCode < 48 || event.keyCode > 57 || event.keyCode != 8){
-        alert ("Only CHARACTERS no Letters");
-    }
-    
+char.addEventListener("keyup", function(keyPress) {
+	var newRows = rows.value;
+	var newChar = char.value;
+	event.preventDefault();
+    	
+    if(keyPress.keyCode !== 9 && keyPress.keyCode !== 8 && keyPress.keyCode !== 16 && keyPress.keyCode !== 49 && keyPress.keyCode !== 50 && keyPress.keyCode !== 51 && keyPress.keyCode !== 52 && keyPress.keyCode !== 53 && keyPress.keyCode !== 54 && keyPress.keyCode !== 55 && keyPress.keyCode !== 56 && keyPress.keyCode !== 57&& keyPress.keyCode !== 58 && keyPress.keyCode !== 59 && keyPress.keyCode !== 13) {
+        alert("dont");
+        
+}
+    else if (newRows.length > 1 || newChar.length > 1) {
+    	alert("ONLY ONE CHARACTER");
 
+	}
+	 else if ( event.keycode === 9){
+	 	alert("only one special char")
+	 }
+   
 });
 
+form.addEventListener("keyup", function(keyPress){
+	if(keyPress.keyCode === 13){
+		document.getElementById('submit').click();
+	}
+})
+
+
+
+
 
 
 
```

Comment:
For future reference, make sure you don't leave empty lines at the end of your file. Best practice is a single blank line. There's a Sublime setting to assure that.